kuwala/core/canvas/src/utils/TransformationCatalogUtils.js

```diff
@@ -0,0 +1,29 @@
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {
+    faAlignLeft,
+    faCalculator,
+    faClock,
+    faFilter,
+    faLink,
+    faMapMarkedAlt,
+    faShuffle
+} from "@fortawesome/free-solid-svg-icons";
+
+export const getCatalogItemIcon = (iconName) => {
+    switch (iconName){
+        case 'clock':
+            return <FontAwesomeIcon icon={faClock}/>
+        case 'align-left':
+            return <FontAwesomeIcon icon={faAlignLeft}/>
+        case 'calculator':
+            return <FontAwesomeIcon icon={faCalculator}/>
+        case 'map-marked-alt':
+            return <FontAwesomeIcon icon={faMapMarkedAlt}/>
+        case 'link':
+            return <FontAwesomeIcon icon={faLink}/>
+        case 'filter':
+            return <FontAwesomeIcon icon={faFilter}/>
+        default:
+            return <FontAwesomeIcon icon={faShuffle}/>
+    }
+}
```

Comment:
I think we don't need this entire file because you can use the icon id directly with the property `icon` of the `FontAwesomeIcon` component. See their documentation: https://fontawesome.com/v5/docs/web/use-with/react#icon-syntax

This here actually causes issues and is not really maintainable. Already now, not all icons are mapped properly that I send from the backend. For example the `general` category has three different icons but as of now, in the frontend the default icon from here (shuffle) is returned.

![image](https://user-images.githubusercontent.com/28758040/165483504-80ca8fab-55dd-4a48-adbf-686663280241.png)

I always return the ids how they are used in FontAwesome for the icons so it should be all working directly.

Reply:
Currently solved using a global config as mentioned in link above, we still need to specify which icon going to be used by using `library.add(iconName)` but it is still much better than current function though.

There is much better way to implement it using `Dynamic Import` by implementing babel macros
https://fontawesome.com/docs/web/use-with/react/add-icons#dynamic-icon-importing

I have tried to and failed to implement the dynamic import, i encounter this problem 
https://stackoverflow.com/questions/69472624/font-awesome-6-beta-and-dynamic-icon-name-in-next-js

there might be more to configure in the babel side. therefore for now i will be using the global config.